doc/man7/provider-cipher.pod

```diff
@@ -313,6 +313,14 @@ Gets a implementation specific randomly generated key for the associated
 cipher ctx. This is currently only supported by 3DES (which sets the key to
 odd parity).
 
+=item "alg_id_param" (B<OSSL_CIPHER_PARAM_ALG_ID>) (octet string)
+
+Used to pass the DER encoded AlgorithmIdentifier parameter to or from
+the cipher implementation.  For most algorithms, where this DER encoded
+parameter has a standard form (it's only an octet string with the IV),
+this is not needed.  Such algorithms should have the flag
+B<EVP_CIPH_FLAG_DEFAULT_ASN1>.
```

Comment:
I think the move to providers gives us the opportunity to swap the defaults the other way around, i.e. if you don't specify a flag then you get this behaviour. We probably need a new flag instead, e.g. something like `EVP_CIPH_FLAG_NON_DEFAULT_ASN1`

In other words, you get default ASN1 behaviour if:

1) You have a legacy cipher and `EVP_CIPH_FLAG_DEFAULT_ASN1` is set
2) You have a provided cipher and `EVP_CIPH_FLAG_NON_DEFAULT_ASN1` is *not* set

Reply:
Yeahok.  But I think `EVP_CIPH_FLAG_CUSTOM_ASN1` is a better name

Reply:
Done